### PR TITLE
JBPM-5903: Stunner - UXD palette review.

### PR DESCRIPTION
**Pull request by zannino**
hello @romartin @pefernan this is the commit for the palette upgrade. See this [ticket](https://issues.jboss.org/browse/JBPM-5903) .
Thanks
Federico

### CODE REVIEW

**Comment by romartin** [comment]
ok to test

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by romartin** [comment]
Hey @zannino ,
After today's review, latest status about this is that we're missing two things:
- Remove horizontal scroll-bars
- Make the palette floating on top of the canvas and make it visible even if scrolling. So the idea is to remove the grey area around the canvas and put the palette on the canvas' top left position. If the canvas is big and there exist scroll bars, the palette must be visible even when scrolling too.
Please, can you update this PR with these new changes and let me know? 
Thanks in advance!

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/BS3PaletteWidgetViewImpl.css [hunk 1]
```diff
@@ -60,22 +59,25 @@
...
+    color: #393f44;
+    font-size: calc(21px);
     margin: auto;
     min-width: calc(27.999996px);
+    text-align: center;
 }
 
 .kie-palette .list-group-item:hover {
     background-color: #fff;
+    /*border-top: 1px solid #bbb;*/
```
Comment (karreiro) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/BS3PaletteWidgetViewImpl.css]: I think that you can remove this line.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/BS3PaletteWidgetViewImpl.css [hunk 2]
```diff
@@ -135,54 +137,67 @@
...
+    margin-bottom: calc(12px);
+    padding-left: calc(18px);
 }
 
 .kie-palette-flyout {
     background-color: #fff;
     border-color: #bbb;
-    border: none;
-    left: calc(90px);
+    /*border: none;*/
```
Comment (karreiro) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/palette/BS3PaletteWidgetViewImpl.css]: I think that you can remove this line, too.

**Comment by zannino** [comment]
Hello @romartin , i updated everything you told me to fix.

**Comment by romartin** [comment]
hey @hasys ! Thanks for the review.

You're right about the palette now overlaps the start event. In fact I said to @zannino to move forward, but yeah, looking at the screenshot now it makes sense trying to fix this as we cannot ensure that this will not happen when opening existing diagrams from other editors. I'm adding some comments about this [in the jira ticket](https://issues.jboss.org/browse/JBPM-5903), let's see if UX can help.

About second point - canvas scrolling - you're right too @hasys , but this is not now feasible for @zannino to fix in this commit. This will be addresses as soon as starting [this ticket](https://issues.jboss.org/browse/JBPM-5536). The problem is the  the canvas size is actually hardcoded and cannot be resized yet....

**Comment by hasys** [comment]
Hi @romartin,
for me palette overlaps not only Start event but all blue area on screenshot. So I am able to select NO one element. If I will drag and drop new element from palette to the canvas it will appear but I will be blocked to do anything with it. The only option is place new node out of Blue area.

To see this area you can use Google Chrome tools (Ctrl+Shift+C). Just try to move you mouse over the nodes using this tool.

So for me it is blocker. Ping me if you are not able to reproduce this behavior and you are able to select any node on the canvas.

**Comment by romartin** [comment]
ahhh okk I see @hasys , thanks for the update!! so yeah, this is a bug to fix for @zannino :)

**Comment by romartin** [comment]
Hey @zannino @hasys , so missing these 2 things:
- Palette overlaps all horizontal area (on top of canvas) - > bug to fix by @zannino 
- Wait for some UX feedback about the current palette position and elements that could be below it. Added comments i[n the ticket](https://issues.jboss.org/browse/JBPM-5903).
Agree? :) 
Thanks!

**Comment by hasys** [comment]
Hi @romartin, yes, sounds good :)

**Comment by zannino** [comment]
Hi @romartin  @hasys . i fixed tthe bug Palette overlaps all horizontal area (on top of canvas).

Federico #uberfire F3deL

**Comment by romartin** [comment]
hey @zannino , well yes, the bug is fixed, but the palette is not floating on top of canvas now....

**Comment by romartin** [comment]
hey @zannino 

You broked as well the standalone showcase - the toolbar does not appears complete! 

screenshot from 2017-05-10 21-07-05

Please verify all before pushing, this takes time for all to build....

**Comment by romartin** [comment]
So again, missing things:
- Palette is not on top of canvas
- The toolbar is not visible on the standalone showcase

**Comment by romartin** [comment]
ok to test

**Comment by zannino** [comment]
hey @romartin 
added new commit that fixes the toolbar is not visible on the standalone showcase

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by zannino** [comment]
Hi @romartin, 
i fixied the bug.
I inserted the property  in the css file.

Federico

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins please retest this

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/presenters/session/impl/SessionPresenterView.html [hunk 3]
```diff
@@ -30,3 +32,35 @@
         </td>
     </tr>
 </table>
+
+-->
+
+<table>
+    <tr>
+        <td>
+            <span data-field="loadingPanel" class="label label-danger loading-panel">Loading</span>
```
Comment (karreiro) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/presenters/session/impl/SessionPresenterView.html]: I think that there's a missing i18n here.

**Comment by romartin** [comment]
Hey @karreiro ,
Thanks for the feedback!! Well seen :) 
Anyway, we'll address that missing I18n in he next iterations as with [this ticket](https://issues.jboss.org/browse/JBPM-5622), please could you mind approving this ticket considering that we'll fix this literal later? We need the new palette design merged as for tomorrow, as we have a presentation with the UX guys and would be nice showing this.
Thanks!!

**Comment by karreiro** [comment]
Approved @romartin ;-)

**Comment by romartin** [comment]
thanks @karreiro @ederign !!!

**Pull request merged**